docs: [vision-on-sample-app][6/n] AppDelegate

Ahmed-CIO opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* (to be filled)

## Context

This is a PR in a series of PRs to build sample app that runs Identify and Track in VisionPro
The app is built to be an interactive walk through guidance for developers who are integrating Customer.io Swift SDK in their VisioPro apps.
For more context:

- See the project [one pager](https://www.notion.so/custio/Spacial-Identify-and-Tracking-6587d848bbde494095c0585237326ed3?pvs=4)
- See [Linear Issue](https://linear.app/customerio/issue/MBL-115/sample-app-for-visionpro)

## PR Summary
This PR introduces fthe app delegate which will be used to initialize `CustomerIO`

## Test plan
- Open `VisionOs.xcproject`
- Hit CMD+R to run the project and make sure you have VisionOS simulator installed and selected

levibostian commented on Apps/VisionOS/VisionOS/AppDelegate.swift:
```diff
@@ -0,0 +1,39 @@
+    ) -> Bool {
+        /*
+
+         How do I get the values for the siteId and apiKy?
+         1- Login or Signup in CustomerIO platform: https://fly.customer.io/
+         2- On the top right section click the
+         Settings Icon >> Workspace Settings >> API and webhook credentials
+         You will see list of one or more workspaces and each has
+         its own pair of siteId and apiKey, these are the ones
+         to use here.
```
I do greatly appreciate your efforts to help customers find the information that they need in this code! 

I am concerned of this documentation getting out-of-date. Instead, do we have a webpage in our CIO docs site that we could point to here?

Ahmed-CIO replied:
I had similar comment earlier about the fact that the proximity to the code should give it better chance of not being outdated; with that said, a link here to a full document webpage won't hurt. On it

levibostian commented on Apps/VisionOS/VisionOS/AppDelegate.swift:
```diff
@@ -0,0 +1,39 @@
+         */
+
+        let workspaceSettings = AppState.shared.workspaceSettings
+        if workspaceSettings.isSet() {
```
What happens if workspace settings are *not* set? 

At a minimum, should we create a log warning or something?

Ahmed-CIO replied:
Not really. The goal here is not to ensure full setup ready for every app run; the gaol is to ensure the user don't have to go to the initialize screen and call initialize manually. More of a UX piece instead of production-ready setup. At the end of the day, it is a sample app meant to make it easy to see the SDK usage in action instead of being in-itself a production ready app

Ahmed-CIO replied:
But one thing I found missing while working on this part, is that the callsite have noway to tell if they enter wrong credentials. The SDK won't report anything, as far as I know at least. Something we might want to revise at some point later.

Ahmed-CIO commented:
Apologies for that, gh-stack pushes seems to cause too much noise due to the fact that it pushes the entire stack for every push. These changes will carry on #621 

Meanwhile, I addressed the comments in this PR
